common/src/main/java/com/microsoft/hydralab/performance/parsers/AndroidMemoryInfoResultParser.java

```diff
@@ -40,18 +41,37 @@ public PerformanceTestResult parse(PerformanceTestResult performanceTestResult)
             return null;
         }
         List<PerformanceInspectionResult> inspectionResults = performanceTestResult.performanceInspectionResults;
-        for (PerformanceInspectionResult inspectionResult : inspectionResults) {
+        double[] averageMemoryInfo = new double[MEM_INFO_LENGTH];
+        Arrays.fill(averageMemoryInfo, 0.0);
+        for (int i = 0; i < inspectionResults.size(); i++) {
+            PerformanceInspectionResult inspectionResult = inspectionResults.get(i);
             File logFile = inspectionResult.rawResultFile;
             long[] memInfos = parseRawResultFile(logFile);
             inspectionResult.parsedData = buildMemoryInfo(inspectionResult.inspection.appId, inspectionResult.inspection.description, inspectionResult.timestamp, memInfos);
+            updateAverageMem(averageMemoryInfo, memInfos, i);
         }
 
+        performanceTestResult.setResultSummary(buildAverageMemoryInfo(averageMemoryInfo, inspectionResults.get(0)));
+
         return performanceTestResult;
     }
 
+    private void updateAverageMem(double[] averageMemoryInfo, long[] memInfoArray, int index) {
+        for (int i = 0; i < MEM_INFO_LENGTH; i++) {
+            averageMemoryInfo[i] = averageMemoryInfo[i] * index / (index + 1) + (double) memInfoArray[i] / (index + 1);
+        }
+    }
+
+    private AndroidMemoryInfo buildAverageMemoryInfo(double[] averageMemoryInfo, PerformanceInspectionResult inspectionResult) {
+        long[] averageMemoryInfoLong = new long[averageMemoryInfo.length];
+        for (int i = 0; i < averageMemoryInfo.length; i++) {
+            averageMemoryInfoLong[i] = Double.valueOf(averageMemoryInfo[i]).longValue();
```

Comment:
Using Math.round to round will be more accurate

Reply:
Fixed